### DISCUSSION TITLE
perform batch transaction using aptos react wallet provider

### DISCUSSION THREAD

**Discussion by sweetim** [aptos-labs/aptos-developer-discussions#164] [Questions] [answered]
### Describe your question in detail.

how can i perform batch transaction using aptos react wallet provider?
something like in the aptos-ts-sdk
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/examples/typescript/batch_mint.ts

### What have you tried or looked at?  Or how can we reproduce the error?

i read through the source code of aptos wallet provider, it seems it is not possible right now?
https://github.com/aptos-labs/aptos-wallet-adapter

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment, marked as answer]
At the moment, if you want to do multiple actions at once, I would look into either a script or deploying a module that adds multiple items.  Generally wallets work around the idea of submitting one transaction at a time, and handling the response of it afterwards.

https://aptos.dev/en/build/smart-contracts/book/modules-and-scripts#scripts

**Reply by mshakeg** [reply]
Hi @gregnazario is there a batch request feature on the Aptos RPC? would like to make multiple view calls in the same request instead of multiple concurrent requests.

https://docs.infura.io/api/networks/ethereum/how-to/make-batch-requests
